roles/rsyslog/tasks/main.yml

```diff
@@ -130,37 +130,80 @@
       when: (rsyslog_enabled | bool)
       notify: restart rsyslogd
 
-    - name: Run deploy input sub-roles configs
-      include_role:
-        name: "{{ role_path }}/roles/input_roles/{{ input_item.type }}"
+    - name: Include input sub-vars 
+      include_vars:
+        file: "{{ varfile }}"
+      vars:
+        varfile: "inputs/{{ input_item.type }}.yml"
+        varfile_abspath: "{{ role_path }}/vars/{{ varfile }}"
       loop: "{{ logging_inputs | flatten(1) }}"
       loop_control:
         loop_var: input_item
       when:
         - rsyslog_enabled | bool
         - input_item | d([])
-        - input_item.state | d('present') != 'absent'
+        - varfile_abspath is exists
+      tags:
+        - skip_ansible_lint
+
+    - name: Run input sub-tasks 
+      include_tasks:
+        file: "{{ tasks }}"
+      vars:
+        tasks: "inputs/{{ input_item.type }}.yml"
+        tasks_abspath: "{{ role_path }}/tasks/{{ tasks }}"
+      loop: "{{ logging_inputs | flatten(1) }}"
+      loop_control:
+        loop_var: input_item
+      when:
+        - rsyslog_enabled | bool
+        - input_item | d([])
+        - tasks_abspath is exists
+      tags:
+        - skip_ansible_lint
 
-    - name: Run remove input sub-roles configs
-      include_role:
-        name: "{{ role_path }}/roles/input_roles/{{ input_item.type }}"
-        tasks_from: cleanup.yml
+    - name: Run remove input sub-tasks
+      include_tasks:
+        file: "{{ tasks }}"
+      vars:
+        tasks: "inputs/{{ input_item.type }}_cleanup.yml"
+        tasks_abspath: "{{ role_path }}/tasks/{{ tasks }}"
       loop: "{{ logging_inputs | flatten(1) }}"
       loop_control:
         loop_var: input_item
       when:
         - input_item | d([])
         - input_item.state | d('present') == 'absent'
 
-    - name: Run output sub-roles configs
-      include_role:
-        name: "{{ role_path }}/roles/output_roles/{{ output_item.type }}"
+    - name: Include output sub-vars 
+      include_vars:
+        file: "{{ varfile }}"
+      vars:
+        varfile: "outputs/{{ output_item.type }}.yml"
+        varfile_abspath: "{{ role_path }}/vars/{{ varfile }}"
+      loop: "{{ rsyslog_outputs | flatten(1) }}"
+      loop_control:
+        loop_var: output_item
+      when:
+        - rsyslog_enabled | bool
+        - output_item | d([])
+        - varfile_abspath is exists
+      tags:
+        - skip_ansible_lint
+
+    - name: Run output sub-tasks 
+      include_tasks:
+        file: "{{ tasks }}"
+      vars:
+        tasks: "outputs/{{ output_item.type }}.yml"
+        tasks_abspath: "{{ role_path }}/tasks/{{ tasks }}"
       loop: "{{ rsyslog_outputs | flatten(1) }}"
       loop_control:
         loop_var: output_item
       when:
         - rsyslog_enabled | bool
         - output_item | d([])
+        - tasks_abspath is exists
```

Comment:
is it needed? Are there outputs which do not have a task file?

Reply:
Ok...  So, you are suggesting to get rid of {tasks,varfile}_abspath completely...

Reply:
yes, if there is an input or output which lacks a var file or task file, it is probably simpler to add a dummy empty one.